presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusSplit.java

```diff
@@ -299,4 +300,44 @@ public void testCorrectNumberOfSplitsCreated()
         int numSplits = splits.getNextBatch(NOT_PARTITIONED, NUMBER_MORE_THAN_EXPECTED_NUMBER_SPLITS).getNow(null).getSplits().size();
         assertEquals(numSplits, timeUnitsToSeconds(config.getMaxQueryRangeDuration()) / timeUnitsToSeconds(config.getQueryChunkSizeDuration()));
     }
+
+    @Test
+    public void testGeneratedSplitTimesCorrect()
+            throws Exception
+    {
+        String maxQueryRangeDurationStr = "21d";
+        String queryChunkSizeDurationStr = "1d";
+        long nowTimeMillis = Instant.now().toEpochMilli();
+        long nowSeconds = TimeUnit.MILLISECONDS.toSeconds(nowTimeMillis);
+        long maxQueryRangeDuration = timeUnitsToSeconds(maxQueryRangeDurationStr);
+        long queryChunkSizeDuration = timeUnitsToSeconds(queryChunkSizeDurationStr);
+        List<String> splitTimes = PrometheusSplitManager.generateTimesForSplits(nowTimeMillis, maxQueryRangeDurationStr, queryChunkSizeDurationStr);
+        assertEquals(splitTimes.size(), 21);
```

Comment:
Can we just assert what the list itself should look like instead of asserting these other calculations? It should be something we can just do with basic math, which we could explain in a comment.

Reply:
Wouldn't that just be the math encapsulated in `generateTimesForSplits()`? I thought spot-checking it with simpler math that checked its bounds and corner-cases (like where duration isn't divisible evenly by chunk size) were better tests, idk...

Reply:
Yes, if we know the algorithm, doing the math and writing the list of times by hand should be trivial. The point of tests isn't to do the math again outside the method, it's to verify the behavior. In fact, by doing more logic/math in the tests, we now have more logic in the tests which could be wrong.

Reply:
Ah, okay. I made the split timing tests explicitly just the lists of string literals as calculated out-of-band, in 66f40d4